### DISCUSSION TITLE
Entity with wild card schema won't query to related tables in other schemas defined in entities with wild card as well

### DISCUSSION THREAD

**Discussion by josemontessanchez** [mikro-orm/mikro-orm#4770] [General]
*Greetings, im not used to report stuff so forgive me if there is something wrong on this report. Also it may be that we are using incorrectly something apologies if so.*

**Describe the bug**
There seems to be an issue using wild card schema in entities that are related to other tables found in other schemas, with entities using as well wild card schema, preexisting relationship. It won't fetch relations nor data.

Entities are related to each other, each entity exists within a different schema. Entities are using wildcard (its needed as its just an example of needed functionality between schemas).

When query is made using the "customer repository", injecting the corresponding "branch schema" at the branch level, it injects instead the same schema name in the related table at the company level (company schema , in the relationship) instead of fetching the corresponding schema name of related table.

**To Reproduce**

**1. Database:**
-----
![basic-db](https://user-images.githubusercontent.com/70652394/154367027-c56c56b0-039b-422d-85d2-d5317f8f6026.png)

**Company Schema
Schema: company_a**

Company             
+++++++++
     **Field     │ Type**    
     uuid      │ string  
     name      │ string  
    website    |  string
     email     │ string  
     createdAt │ date    
     updatedAt │ date    

**Branch Schema 1
Schema: company_a_branch_b**

Customer            
+++++++
     **Field     │ Type**    
     uuid      │ string  
     name      │ string  
     email     │ string  
     idCompany │ string  
     createdAt │ date    
     updatedAt │ date  
 

**Branch Schema 2
Schema: company_a_branch_f**

Customer            
+++++++
     **Field     │ Type**    
     uuid      │ string  
     name      │ string  
     email     │ string  
     idCompany │ string  
     createdAt │ date    
     updatedAt │ date  

**Relationships:** 1:N companyschema.company.uuid --> branchschema.customer.idCompany

**2. Entities:**
-----

**ABSTRACT ENTITY**

```
import { PrimaryKey, Property } from "@mikro-orm/core";

export abstract class AbstractEntity {
   @PrimaryKey({ type : 'uuid', defaultRaw : 'uuid_generate_v4()' })
   uuid! : string;
  
   @Property()
   createdAt = new Date();

   @Property({ onUpdate : () => new Date() })
   updatedAt = new Date();
}
```

**COMPANY ENTITY at COMPANY schema**

```
import { Entity, Property } from '@mikro-orm/core';
import { AbstractEntity } from '../abstract.entity';

@Entity({ schema : '*' })
export class Company extends AbstractEntity {
   @Property()
   name : string;

   @Property()
   website : string;

   @Property()
   email : string;

   @Property()
   active : boolean;
}
```

**CUSTOMER ENTITY at BRANCH{N} schema**

```
import { Entity, Property, ManyToOne } from '@mikro-orm/core';
import { AbstractEntity } from '../abstract.entity';
import { Company } from '../company.entity';

@Entity({ schema : '*' })
export class Customer extends AbstractEntity{
   @Property()Ty
   name : string;

   @Property()
   email : string;
   
   @ManyToOne({ entity : () => Company })
   idCompany : Company;
}
```

**QUERY**

```
import { InjectRepository } from '@mikro-orm/nestjs';
import { EntityRepository, EntityManager } from '@mikro-orm/postgresql';
import { Injectable } from '@nestjs/common';

import { CustomerDto } from './dto/cliente.dto';
import { Customer } from '../../database/models/customer.entity';

@Injectable()
export class CustomerService {
  
   constructor(
       @InjectRepository(Customer) private readonly customerRepo : EntityRepository<Customer>
  ){}

   async getCutomers( schema : string ) : Promise<Customer[]> {
       return await this.customerRepo.findAll({ populate : true , schema  });
   }

}
```

**3. Executing query**
----
When executing query Mikro-orm takes the same schema name to look up the relationship.
●	company_a_branch_b.customer  (table where i made the query)
●	company_a_branch_b.company  (table reference fk, incorrect)

How can one pass specific schema that relation belongs to, within the populate/findAll/similar while using wild card in entity definition?
●	company_a_branch_b.customer  (table where i made the query)
●	company_a.company  (table reference fk with their respective schema, correct)

If the schema name is defined as static in the company entity, query works correctly:

```
@Entity({ schema : 'company_a' })

export class Company extends AbstractEntity{
```

Result:

> Content-Type: application/json; charset=utf-8
> Content-Length: 379
> ETag: W/"17b-UrC+t+4kg6OdTli8LSOpZYut+7k"
> Date: Wed, 16 Feb 2022 17:51:16 GMT
> Connection: close
> 
> [
> 
>   {
> 
>     "uuid": "9408351a-12ba-4d62-ad5e-884dcd9f1554",
>     "createdAt": "2022-02-16T17:50:37.000Z",
>     "updatedAt": "2022-02-16T17:50:37.000Z",
>     "name": "Dave",
>     "email": "dave@test.com",
>     "idCompany": {
> 
>       "uuid": "fc9f0d89-a4b7-40f6-95ac-0fca2a35ddc8",
>       "createdAt": "2022-02-16T17:49:50.000Z",
>       "updatedAt": "2022-02-16T17:49:50.000Z",
>       "name": "Intek",
>       "website": "intek.com",
>       "email": "intek@test.com",
>       "active": true
>     }
>   }
> ]

If the schema name is defined as wild card in the company entity, query works incorrectly:

```
@Entity({ schema : '*' })

export class Company extends AbstractEntity{
```

Result:
Returns nothing or Error 500

**Stack trace**

```
[Nest] 33019  - 16/02/2022 12:15:13   ERROR [ExceptionsHandler] select "c0".* from "branch_c9c6704c_9978_41b5_aa24_80fd11c09c2a"."company" as "c0" where "c0"."uuid" in ('fc9f0d89-a4b7-40f6-95ac-0fca2a35ddc8') order by "c0"."uuid" asc - no existe la relación «branch_c9c6704c_9978_41b5_aa24_80fd11c09c2a.company»
TableNotFoundException: select "c0".* from "branch_c9c6704c_9978_41b5_aa24_80fd11c09c2a"."company" as "c0" where "c0"."uuid" in ('fc9f0d89-a4b7-40f6-95ac-0fca2a35ddc8') order by "c0"."uuid" asc - no existe la relación «branch_c9c6704c_9978_41b5_aa24_80fd11c09c2a.company»
    at PostgreSqlExceptionConverter.convertException (/home/rafael/Vídeos/mikro-multitenants/node_modules/@mikro-orm/postgresql/PostgreSqlExceptionConverter.js:36:24)
    at PostgreSqlDriver.convertException (/home/rafael/Vídeos/mikro-multitenants/node_modules/@mikro-orm/core/drivers/DatabaseDriver.js:180:54)
    at /home/rafael/Vídeos/mikro-multitenants/node_modules/@mikro-orm/core/drivers/DatabaseDriver.js:184:24
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at PostgreSqlDriver.find (/home/rafael/Vídeos/mikro-multitenants/node_modules/@mikro-orm/knex/AbstractSqlDriver.js:46:24)
    at SqlEntityManager.find (/home/rafael/Vídeos/mikro-multitenants/node_modules/@mikro-orm/core/EntityManager.js:105:25)
    at EntityLoader.populateMany (/home/rafael/Vídeos/mikro-multitenants/node_modules/@mikro-orm/core/entity/EntityLoader.js:144:22)
    at EntityLoader.populateField (/home/rafael/Vídeos/mikro-multitenants/node_modules/@mikro-orm/core/entity/EntityLoader.js:213:25)
    at EntityLoader.populate (/home/rafael/Vídeos/mikro-multitenants/node_modules/@mikro-orm/core/entity/EntityLoader.js:37:13)
    at SqlEntityManager.find (/home/rafael/Vídeos/mikro-multitenants/node_modules/@mikro-orm/core/EntityManager.js:122:9)

previous error: select "c0".* from "branch_c9c6704c_9978_41b5_aa24_80fd11c09c2a"."company" as "c0" where "c0"."uuid" in ('fc9f0d89-a4b7-40f6-95ac-0fca2a35ddc8') order by "c0"."uuid" asc - no existe la relación «branch_c9c6704c_9978_41b5_aa24_80fd11c09c2a.company»
    at Parser.parseErrorMessage (/home/rafael/Vídeos/mikro-multitenants/node_modules/pg-protocol/src/parser.ts:369:69)
    at Parser.handlePacket (/home/rafael/Vídeos/mikro-multitenants/node_modules/pg-protocol/src/parser.ts:188:21)
    at Parser.parse (/home/rafael/Vídeos/mikro-multitenants/node_modules/pg-protocol/src/parser.ts:103:30)
    at Socket.<anonymous> (/home/rafael/Vídeos/mikro-multitenants/node_modules/pg-protocol/src/index.ts:7:48)
    at Socket.emit (node:events:390:28)
    at addChunk (node:internal/streams/readable:324:12)
    at readableAddChunk (node:internal/streams/readable:297:9)
    at Socket.Readable.push (node:internal/streams/readable:234:10)
    at TCP.onStreamRead (node:internal/stream_base_commons:190:23)
    at TCP.callbackTrampoline (node:internal/async_hooks:130:17)
```

**Expected behavior**
Object returned
```
Content-Type: application/json; charset=utf-8
Content-Length: 379
ETag: W/"17b-UrC+t+4kg6OdTli8LSOpZYut+7k"
Date: Wed, 16 Feb 2022 17:51:16 GMT
Connection: close

[

  {

    "uuid": "9408351a-12ba-4d62-ad5e-884dcd9f1554",
    "createdAt": "2022-02-16T17:50:37.000Z",
    "updatedAt": "2022-02-16T17:50:37.000Z",
    "name": "Dave",
    "email": "dave@test.com",
    "idCompany": {

      "uuid": "fc9f0d89-a4b7-40f6-95ac-0fca2a35ddc8",
      "createdAt": "2022-02-16T17:49:50.000Z",
      "updatedAt": "2022-02-16T17:49:50.000Z",
      "name": "Intek",
      "website": "intek.com",
      "email": "intek@test.com",
      "active": true
    }
  }
]
```

**Additional context**

**Versions**

| Dependency | Version |
| - | - |
| node | 17.2.0 |
| typescript | 2017 |
| mikro-orm | 5.0.1 |
| mikro-orm/postgresql | 5.0.1 |

**Comment by B4nan** [comment]
Can you try just released 5.0.2? There were some fixes in schema support.

**Comment by josemontessanchez** [comment]
@B4nan already tested with most recent 5.0.2 , still happening.


```
[Nest] 13583  - 17/02/2022 09:08:27   ERROR [ExceptionsHandler] select "c0".* from "branch_c9c6704c_9978_41b5_aa24_80fd11c09c2a"."company" as "c0" where "c0"."uuid" in ('fc9f0d89-a4b7-40f6-95ac-0fca2a35ddc8') order by "c0"."uuid" asc - no existe la relación «branch_c9c6704c_9978_41b5_aa24_80fd11c09c2a.company»
TableNotFoundException: select "c0".* from "branch_c9c6704c_9978_41b5_aa24_80fd11c09c2a"."company" as "c0" where "c0"."uuid" in ('fc9f0d89-a4b7-40f6-95ac-0fca2a35ddc8') order by "c0"."uuid" asc - no existe la relación «branch_c9c6704c_9978_41b5_aa24_80fd11c09c2a.company»
    at PostgreSqlExceptionConverter.convertException (/home/rafael/Vídeos/mikro-multitenants/node_modules/@mikro-orm/postgresql/PostgreSqlExceptionConverter.js:36:24)
    at PostgreSqlDriver.convertException (/home/rafael/Vídeos/mikro-multitenants/node_modules/@mikro-orm/core/drivers/DatabaseDriver.js:180:54)
    at /home/rafael/Vídeos/mikro-multitenants/node_modules/@mikro-orm/core/drivers/DatabaseDriver.js:184:24
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at PostgreSqlDriver.find (/home/rafael/Vídeos/mikro-multitenants/node_modules/@mikro-orm/knex/AbstractSqlDriver.js:46:24)
    at SqlEntityManager.find (/home/rafael/Vídeos/mikro-multitenants/node_modules/@mikro-orm/core/EntityManager.js:105:25)
    at EntityLoader.populateMany (/home/rafael/Vídeos/mikro-multitenants/node_modules/@mikro-orm/core/entity/EntityLoader.js:144:22)
    at EntityLoader.populateField (/home/rafael/Vídeos/mikro-multitenants/node_modules/@mikro-orm/core/entity/EntityLoader.js:213:25)
    at EntityLoader.populate (/home/rafael/Vídeos/mikro-multitenants/node_modules/@mikro-orm/core/entity/EntityLoader.js:37:13)
    at SqlEntityManager.find (/home/rafael/Vídeos/mikro-multitenants/node_modules/@mikro-orm/core/EntityManager.js:122:9)

previous error: select "c0".* from "branch_c9c6704c_9978_41b5_aa24_80fd11c09c2a"."company" as "c0" where "c0"."uuid" in ('fc9f0d89-a4b7-40f6-95ac-0fca2a35ddc8') order by "c0"."uuid" asc - no existe la relación «branch_c9c6704c_9978_41b5_aa24_80fd11c09c2a.company»
    at Parser.parseErrorMessage (/home/rafael/Vídeos/mikro-multitenants/node_modules/pg-protocol/src/parser.ts:369:69)
    at Parser.handlePacket (/home/rafael/Vídeos/mikro-multitenants/node_modules/pg-protocol/src/parser.ts:188:21)
    at Parser.parse (/home/rafael/Vídeos/mikro-multitenants/node_modules/pg-protocol/src/parser.ts:103:30)
    at Socket.<anonymous> (/home/rafael/Vídeos/mikro-multitenants/node_modules/pg-protocol/src/index.ts:7:48)
    at Socket.emit (node:events:390:28)
    at addChunk (node:internal/streams/readable:324:12)
    at readableAddChunk (node:internal/streams/readable:297:9)
    at Socket.Readable.push (node:internal/streams/readable:234:10)
    at TCP.onStreamRead (node:internal/stream_base_commons:190:23)
    at TCP.callbackTrampoline (node:internal/async_hooks:130:17)
```

Wildcard schema is not considered altough relationships are already there, seems there is a need to inject 2 different schemas in query with wild card schemas,
or mikro-orm needs to fetch schema/table relations already in place for the table specified in the main query entity.

As stated before if using wild card schema in just one of the entities, query works completely fine, if using wild card schema on both entities it fails as it just takes one schema definition instead of fetching relationships to whatever schema/tables needed.

Greets.

**Comment by B4nan** [comment]
I can't reproduce this. Here is a passing test case:

```ts
import { PrimaryKey, Entity, Property, ManyToOne, MikroORM, wrap } from '@mikro-orm/core';

@Entity({ schema: '*' })
export class Company {

  @PrimaryKey()
  id!: number;

  @Property()
  name!: string;

}

@Entity({ schema: '*' })
export class Customer {

  @PrimaryKey()
  id!: number;

  @Property()
  name!: string;

  @ManyToOne({ entity: () => Company })
  company!: Company;

}

describe('GH issue 2776', () => {

  let orm: MikroORM;

  beforeAll(async () => {
    orm = await MikroORM.init({
      entities: [Customer, Company],
      dbName: `mikro_orm_test_gh_2776`,
      type: 'postgresql',
    });
    await orm.getSchemaGenerator().refreshDatabase();
    await orm.getSchemaGenerator().updateSchema({ schema: 's1' });
    await orm.em.nativeDelete(Customer, {}, { schema: 's1' });
  });

  afterAll(() => orm.close(true));

  test('wildcard entities', async () => {
    const c = new Customer();
    c.name = 'e';
    c.company = new Company();
    c.company.name = 'c';
    wrap(c).setSchema('s1');
    wrap(c.company).setSchema('s1');
    await orm.em.fork().persistAndFlush(c);
    const res = await orm.em.getRepository(Customer).findAll({ populate: true, schema: 's1' });
    expect(res).toHaveLength(1);
    expect(wrap(res[0].company).isInitialized()).toBe(true);
    expect(res[0].name).toBe('e');
    expect(res[0].company.name).toBe('c');
  });

});
```

Note that I removed most of the unrelated code like scalar properties. 

Please provide failing repro, ideally by modifying this test case.

**Comment by josemontessanchez** [comment]
Greetings @B4nan 
Tested following structure provided on previous post.

Schema 1 = s1
Schema 2 = s2

Company in schema s1
Customer in schema s2

Related by
```
ALTER TABLE s2.customer
                  ADD CONSTRAINT fk_s1company_s2customer
                  FOREIGN KEY(company_id)
                  REFERENCES s1.company(id);
```

----

Tried to specify both schemas and resulted on failure, it still wants to only get one schema for relations and definition. 
If forcing entity definition with static schema instead wildcard schema it works correctly.
May seem what is needed is to specify both schemas (1 for company, 1 for customer) on query o some other place besides the entities, or for mikro to Fetch existing relationships from schema-to-schema. Unless we are using entity or relationship definition incorrectly.

[Made a public repo with case for testing]( https://github.com/jmontes-enthous/test-2776) just needs to create db mikro_orm_test_gh_2776.

Stack reports the following after trying to specify both schemas on which relationships are stablished:
```

> test-2776@0.0.1 test
> jest

 FAIL  src/test-2776/test-2776.service.spec.ts
  Test2776Service
    ✕ wildcard entities (41 ms)

  ● Test2776Service › wildcard entities

    TableNotFoundException: select "c0".* from "s2"."company" as "c0" where "c0"."id" in (1) order by "c0"."id" asc - relationship does not exist «s2.company»

      at PostgreSqlExceptionConverter.convertException (../node_modules/@mikro-orm/postgresql/PostgreSqlExceptionConverter.js:36:24)
      at PostgreSqlDriver.convertException (../node_modules/@mikro-orm/core/drivers/DatabaseDriver.js:180:54)
      at ../node_modules/@mikro-orm/core/drivers/DatabaseDriver.js:184:24
      at PostgreSqlDriver.find (../node_modules/@mikro-orm/knex/AbstractSqlDriver.js:46:24)
      at SqlEntityManager.find (../node_modules/@mikro-orm/core/EntityManager.js:105:25)
      at EntityLoader.populateMany (../node_modules/@mikro-orm/core/entity/EntityLoader.js:144:22)
      at EntityLoader.populateField (../node_modules/@mikro-orm/core/entity/EntityLoader.js:213:25)
      at EntityLoader.populate (../node_modules/@mikro-orm/core/entity/EntityLoader.js:37:13)
      at SqlEntityManager.find (../node_modules/@mikro-orm/core/EntityManager.js:122:9)
      previous error: select "c0".* from "s2"."company" as "c0" where "c0"."id" in (1) order by "c0"."id" asc - no existe la relación «s2.company»
      at Parser.parseErrorMessage (../node_modules/pg-protocol/src/parser.ts:369:69)
      at Parser.handlePacket (../node_modules/pg-protocol/src/parser.ts:188:21)
      at Parser.parse (../node_modules/pg-protocol/src/parser.ts:103:30)
      at Socket.<anonymous> (../node_modules/pg-protocol/src/index.ts:7:48)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 1 total
Snapshots:   0 total
Time:        2.816 s, estimated 4 s
Ran all test suites.
Jest did not exit one second after the test run has completed.

This usually means that there are asynchronous operations that weren't stopped in your tests. Consider running Jest with `--detectOpenHandles` to troubleshoot this issue.
```

**Comment by B4nan** [comment]
Oh so you want to have relations between wildcard schema entities, now I get it. What is the use case? This sounds quite weird and is indeed not supported. You have relation between two wildcard entities without specifying schema explicitly, that means they both live in the same schema and their FKs are defined this way. Haven't tried your full repro, but if I just copypaste that test case to the one I send above, I am not even able to fire the insert query because of this and it totally makes sense. 

If you have some "root" schema, you should not use wildcard entity for that, but rather have separate entity definitions with static schema name. This way you can define the relations in your wild card entities to point to this static schema. I don't see a better way to achieve this, I wouldn't call this a bug, rather a design limitation or misusage. 

edit: or you could still use wildcard entities, but the relation should be defined with explicit schema name, that might work too

**Comment by josemontessanchez** [comment]
>  What is the use case?

In a case on which i do have different companies (lets say 3) and customers belonging to different branches of those companies (lets say 3 branches per company). I need to relate/query customer per branch per company, entities are the same it just changes schema and there is a need to specify schema on query without query builder on each case.

> haven't tried your full repro, but if I just copypaste that test case to the one I send above, I am not even able to fire the insert query because of this and it totally makes sense.

Its similar, but the two tables are defined in two separate schemas, test-2776.service.spec.ts will build the tables and relationship for those 2 schemas.

> If you have some "root" schema, you should not use wildcard entity for that, but rather have separate entity definitions with static schema name.

In the end there is a need for create dinamically N schemas for company and N schemas for branch, therefore specifying static schema for N entities is not possible.
If there could be a way to insert/specify two schemas in query that would solve the issue and would make it so it would work with N schemas related to other schemas using wild card schema therefore allowing dinamic creation of schemas using same entities. Or if there could be a way to fetch existing relationships between schemas that would also work as well.

Greets.

**Comment by B4nan** [comment]
> If there could be a way to insert/specify two schemas in query that would solve the issue and would make it so it would work with N schemas related to other schemas using wild card schema therefore allowing dinamic creation of schemas using same entities. Or if there could be a way to fetch existing relationships between schemas that would also work as well.

Not really, this would solve your current very simple use case, but it is far away from an actual solution. What if there are 3 schemas? And how will you handle mapping? And what if you want to use one schema for relation A and C, but a different schema for relation B? 

You will need a QB for those use cases. I don't think its supported with QB right now, but we can find a way to do it for sure.

On schema level, you might need to remove some FK constraints generated by the ORM (e.g. the one I mentioned above, that forces the relation to be in the same schema), I don't think we can find a better way than allowing to disable FKs for specific relation (so you would disable them right ahead in wildcard entities for those you know should go to a different schema than the owning entity.

**Comment by josemontessanchez** [comment]
Indeed you are correct, if there are more schemas involved specifying just 2 won't make it. :(

So far we were using Query Builder to pass the name of schemas involved onto variables and then injecting those on to query, making sure that same tables/fields do exist in schemas of same level and related by fk. We have indeed 3 different schemas besides Public, 1 per "level" with relationships and queries between each and work fine injecting schema name on variables needed (Eg. Public, Users, Company, Branches). But yup fields, tables and relationships are the same and must be the same on N schemas from same level and even between hierarchic schemas.

Maybe if there was a way to inject those schema name variables to entity definition instead of wildcard schema, and for relationship definition on entities (to other schemas). Trying to mimic somehow what is done by hand on QB.

**Comment by rlsdev13** [comment]
Hi, using the test case I try to inject the schema in the relation with wrap just for 1 record, and after that load the relation but whit the new schema.

```typescript
const [customer] = await orm.em.getRepository(Customer).findAll({ schema : 's2' }); 
wrap(customer.company).setSchema('s1');
await orm.em.populate( customer, ['company'] );
```

But it fails when mikro loads the relationship. Instead I try to load the relationship in another test but only with a schema for both entities. And the test also fails, here is the test case.

```typescript
import { MikroORM, wrap, Entity, PrimaryKey, Property, ManyToOne } from '@mikro-orm/core';

@Entity({ schema : '*' })
export class Author{
  @PrimaryKey()
  id! : number;
  
  @Property()
  name! : string;
}

@Entity({ schema : '*' })
export class Book{
  @PrimaryKey()
  id! : number;

  @Property()
  name! : string;

  @ManyToOne({ entity : () => Author })
  author! : Author;
}

describe('Test', () => {
  
  let orm  : MikroORM;
  

  beforeAll( async () => {
    orm = await MikroORM.init({
      entities : [ Book, Author ],
      type : 'postgresql'
    })

    await orm.getSchemaGenerator().refreshDatabase();
    await orm.getSchemaGenerator().updateSchema({ schema : 'test'});
    await orm.em.nativeDelete(Book,{},{ schema : 'test' });
  });

  afterAll( () => {
    orm.close(true)
  });

  test('populate already loaded entities', async () => {
    const b = new Book();
    b.name = 'b';
    b.author = new Author();
    b.author.name = 'a';
    wrap(b).setSchema('test');
    wrap(b.author).setSchema('test');
    await orm.em.fork().persistAndFlush(b);
    const books = await orm.em.getRepository(Book).findAll({ schema : 'test' });
    await orm.em.populate(books,['author']);//populating already loaded entities

    expect( wrap(books[0].author).isInitialized() ).toBe(true);
    expect(books[0].name).toBe('b');
    expect(books[0].author.name).toBe('a');
  })

});

```

**Reply by B4nan** [reply]
FYI this test case is passing with latest version, at least with v6.
